.github/workflows/qa.yml

```
@@ -0,0 +1,150 @@
+      - id: matrix
+        run: |
+          python3 - <<'PY' >> "$GITHUB_OUTPUT"
+          import json
+          import os
+
+          attempts = int(os.environ["QA_ATTEMPTS"])
+          print(f"matrix={json.dumps(list(range(1, attempts + 1)))}")
+          PY
+
+  qa:
+    name: "${{ env.QA_TEMPLATE }}:${{ env.QA_TARGET }} (#${{ matrix.attempt }})"
```

actionlint (1.7.11)

[error] context "env" is not allowed here. available contexts are "github", "inputs", "matrix", "needs", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)